dll/win32/shlwapi/propbag.cpp

```diff
@@ -1463,7 +1463,7 @@ CViewStatePropertyBag::_GetMRUSlots(
         return hr;
 
     hr = pMruList->QueryPidl(pidl, cSlots, puSlots, pcSlots);
-    if (hr == S_OK && MODE_CAN_WRITE(dwMode))
+    if (hr == S_OK || MODE_CAN_WRITE(dwMode))
```

Comment:
Why that change? Is this a bugfix that would fix anything outside of your PR? If not, why is it needed for your work?

Reply:
This is a bugfix (and the other one further down fixes a crash).  It will not fix anything outside this PR because SHGetViewStatePropertyBag only exists for DefView to get a property storage interface it can use like it does in this PR.

Without this change, nothing will ever be saved. The old code is `if (FoundExistingEntry && CanWrite)` and since nobody has saved before, this will never be true.

I admit this looks somewhat weird. It would be better if we could keep the && and instead check with SUCCEEDED. I will investigate this again to see if CMruPidlList::QueryPidl can return S_FALSE in the case we care about.

Reply:
Thank you for the explanation. I do consider this discussion to be resolved to my satisfaction, but I don't want to collapse your reply.

Reply:
It is possible the bug is somewhere else, @katahiromz might know since I believe he wrote this. Something has to be done, otherwise SHGetViewStatePropertyBag will never allow you to write anything.

Reply:
[bag.patch](https://github.com/user-attachments/files/16170493/bag.patch)

Can you try this patch? Thanks.

Reply:
> Can you try this patch?

If I undo my fix and apply this instead it stops working.

Reply:
Don't undo. Please use both yours and mine.

Reply:
Why? It already works without your patch.

Reply:
Please add a comment as `"FIXME: HACK!"`.